src/core/include/openvino/op/concat.hpp

```diff
@@ -54,10 +54,9 @@ class OPENVINO_API Concat : public Op {
     bool evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const override;
```

Comment:
Please remove this old implementation

Reply:
I've want to remove to not have duplicates but it looks like is not possible now. When there is no method for host tensor the `ov_template_func_tests` will fail, there is exception that operator not support evaluate. It looks like these test direct call evaluate for host tensor version.

Reply:
Can we modify template plugin?
Or it requires a lot of changes?

Reply:
I think  modification of template plugin is part of #13944. 
@artkuli can share opinion how much work it can be to update template plugin regarding testing and dev?

Regarding migration of evaluate method to ov::tensor version it could be some work as there is a lot of operators so I would like to postpone this migration after merge #13944 and this PR. Then migration can start.